src/newgrf_properties.h

```diff
@@ -41,10 +41,10 @@ enum PropertyID {
 	PROP_ROADVEH_SHORTEN_FACTOR                 = 0x23, ///< Shorter vehicles
 
 	PROP_SHIP_COST_FACTOR                       = 0x0A, ///< Purchase cost
-	PROP_SHIP_SPEED                             = 0x0B, ///< Max. speed: 1 unit = 1/3.2 mph = 0.5 km-ish/h
 	PROP_SHIP_CARGO_CAPACITY                    = 0x0D, ///< Capacity
 	PROP_SHIP_RUNNING_COST_FACTOR               = 0x0F, ///< Yearly runningcost
 	PROP_SHIP_CARGO_AGE_PERIOD                  = 0x1D, ///< Number of ticks before carried cargo is aged
+	PROP_SHIP_SPEED                             = 0x23, ///< Max. speed: 1 unit = 1/3.2 mph = 0.5 km-ish/h
```

Comment:
I'm no NewGRF expert, but I fear this change breaks NewGRFs that use `PROP_SHIP_SPEED` as in the NewGRFs that would simply be encoded as `(0x)0B`. In other words, I think the value of the constant may not be changed.
Having said that, I do not think it is a problem to leave it at the old value as the callback will return a 14 or 15 bit value from the NewGRF. That even makes me wonder whether the issue there was with fast ships might have been due to a ship defining 256 units via this callback.

Reply:
Is this enum made available outside of the OpenTTD code? If so then indeed this could lead to problems with existing NewGRFs. I'm no NewGRF expert either, I hope someone with a bit more domain knowledge kan weigh in here.

Are you saying that we could also just change the byte to a uint16 or increase the max allowed speed? I can image that that would give issues with existing NewGRFs, but again, no expert... it might work?

Reply:
`available for callback 0x36` means that NewGRF expect that callback to be run for prop 0B to get the speed (and GRF is all raw numbers, there are no symbolic constants), not prop 23.

Now while there might be existing NewGRFs that wrongly try to return too large nonsense values in CB 36, it might be safe to just use the full callback result. In fact, I think there are things already that just say "use CB 36 if you need a value bigger than prop X allows".